From 323e700ba8e929dc21fb8d776bcedb38fb3340b1 Mon Sep 17 00:00:00 2001
From: Nate Myren <ntmyren@google.com>
Date: Thu, 10 Nov 2022 14:09:34 -0800
Subject: [PATCH] RESTRICT AUTOMERGE Use static token for myAttributionSource
 in ServiceUtilities

Bug: 258672042
Test: manual
Change-Id: I44243bbbd68db9ee8b5c4ba02a061e14bd300d20
(cherry picked from commit 9a213a1ef475d828d02d0055e2b713cf2068c683)
Merged-In: I44243bbbd68db9ee8b5c4ba02a061e14bd300d20
---
 media/utils/ServiceUtilities.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/media/utils/ServiceUtilities.cpp b/media/utils/ServiceUtilities.cpp
index 9c7b863645..04e280a487 100644
--- a/media/utils/ServiceUtilities.cpp
+++ b/media/utils/ServiceUtilities.cpp
@@ -101,7 +101,10 @@ std::optional<AttributionSourceState> resolveAttributionSource(
     AttributionSourceState myAttributionSource;
     myAttributionSource.uid = VALUE_OR_FATAL(android::legacy2aidl_uid_t_int32_t(getuid()));
     myAttributionSource.pid = VALUE_OR_FATAL(android::legacy2aidl_pid_t_int32_t(getpid()));
-    myAttributionSource.token = sp<BBinder>::make();
+    // Create a static token for audioserver requests, which identifies the
+    // audioserver to the app ops system
+    static sp<BBinder> appOpsToken = sp<BBinder>::make();
+    myAttributionSource.token = appOpsToken;
     myAttributionSource.next.push_back(nextAttributionSource);
 
     return std::optional<AttributionSourceState>{myAttributionSource};
-- 
2.39.0.rc1.256.g54fd8350bd-goog

